src/main/scala/io/iohk/praos/domain/BlockFactory.scala

```diff
@@ -0,0 +1,24 @@
+package io.iohk.praos.domain
+
+import io.iohk.praos.crypto._
+
+
+case class BlockFactory(signer: Signer, vrf: VerifiableRandomFunction) {
```

Comment:
Maybe we should rename this to `BlockGenerator` or rename `TransactionsGenerator` to `TransactionsFactory` for consistency? What do you think @mkatixlabs ?

Reply:
In my experience, "Generator" is used more for representing things like RNGs or fake data procedures like `TransactionGenerator`. At least to me, generators and factories are different concepts (in a subtle way, of course).